app/models/pglogical_subscription.rb

```diff
@@ -86,6 +86,18 @@ def pglogical(refresh = false)
     self.class.pglogical(refresh)
   end
 
+  def self.validate(host, port, user, password, dbname, id)
+    return MiqRegionRemote.validate_connection_settings(host, port, user, password, dbname) if id.nil?
+    subscription = find(id)
+    return [_("Subscription record not found")] if subscription.nil?
+    subscription.validate(host, port, user, dbname)
+  end
+
+  def validate(host, port, user, dbname)
```

Comment:
I would make these parameters keyword args so that if I have a subscription object, and I want to validate it as is, I can just call `subscription.validate`